### PR TITLE
Refactor part 1

### PR DESCRIPTION
**Pull request by danmrichards**
First pass at a refactor, targeted at tidying the global state. Next step is to modularise the _roaching_ process

## Changes
* Introduced new radroach struct to manage config.
* Introduced functional options pattern to toggle verbose logging and enum-to-check functionality.
* Refactored the enum to check logic accordingly.
* Removed verboseLogger struct, now part of radroach.
* Refactored logging to use radroach scoped logger instead of global

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -3,90 +3,35 @@ package main
...
 	flag.Parse()
 
 	if len(flag.Args()) < 2 {
 		usage()
 		os.Exit(1)
 	}
 
-	opts.srcMySQL = flag.Args()[0]
-	if len(opts.srcMySQL) == 0 {
+	src := flag.Args()[0]
```
Comment (ladydascalie) [inline main.go]:
```
src := flags.Args()[0]
dst := flags.Args()[1]

if len(src) == 0 || len(dst) == 0 {
	usage()
	os.Exit(1)
}
```

File: radroach.go [hunk 2]
```diff
@@ -92,7 +140,10 @@ func (r *roacher) roach() (output []byte, err error) {
 		*t = constraints.ReplaceAll(*t, empty)
 
 		// Stage 2.2: Enums.
-		processEnums(string(name[1]), t, tableConstraints)
+		if r.enumToCheck {
+			enumsToChecks(string(name[1]), t, tableConstraints)
+		}
+		*t = enum.ReplaceAll(*t, []byte(" TEXT "))
```
Comment (ladydascalie) [inline radroach.go]: `*t` everywhere is pretty confusing.

todo: rework the code so we can just deal with values

File: main.go (hunk 1 above)
Comment (danmrichards) [reply main.go]: Fixed

File: radroach.go (hunk 2 above)
Comment (danmrichards) [reply radroach.go]: Yeah I'm not happy with that either

Comment (danmrichards) [reply radroach.go]: Done

**Review by ladydascalie — APPROVED** [review]

**Pull request merged**